scripts/signup.js

```diff
@@ -1,125 +1,218 @@
-    const passwordError = document.querySelector("#passwordError").textContent;
-    const confirmPasswordError = document.querySelector(
-      "#confirmPasswordError"
-    ).textContent;
-
-    return !emailError && !passwordError && !confirmPasswordError;
+// 비밀번호 입력 처리 함수 : 유효성 검사 및 에러 메시지 관리
+function handlePasswordInput(event) {
+  const passwordValue = passwordInput.value;
+  const confirmPasswordValue = confirmPasswordInput.value;
+
+  // 현재 입력 필드에 따라 에러 메시지 지우기
+  if (event.target === passwordInput) {
+    passwordError.textContent = "";
+    passwordInput.classList.remove(INVALID_CLASS);
+  } else if (event.target === confirmPasswordInput) {
+    confirmPasswordError.textContent = "";
+    confirmPasswordInput.classList.remove(INVALID_CLASS);
   }
 
-  function handleSignUp(event) {
-    event.preventDefault();
-    if (validateSignUp()) {
-      window.location.href = "component/folder.html";
-    }
+  // 초기화
+  passwordError.textContent = "";
+  confirmPasswordError.textContent = "";
+
+  // 비밀번호 입력부분이 비어 있는 경우
+  if (!passwordValue.trim()) {
+    passwordError.textContent = "비밀번호를 입력해 주세요.";
+    passwordInput.classList.add(INVALID_CLASS);
+  }
+  // 비밀번호 입력부분이 비어있지 않는 경우 + 비밀번호 형식이 유효하지 않은 경우
+  else if (!isValidPassword(passwordValue)) {
+    passwordError.textContent =
+      "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.";
+    passwordInput.classList.add(INVALID_CLASS);
+  }
+  // 비밀번호 입력부분이 비어있지 않은 경우 + 비밀번호 형식이 유효한 형식인 경우 + 비밀번호 길이가 8자 이상인 경우
+  else if (passwordValue.length < 8) {
+    passwordError.textContent = "비밀번호는 8자 이상 입력해 주세요.";
+    passwordInput.classList.add(INVALID_CLASS);
+  }
+  // 비밀번호 입력부분이 비어있지 않은 경우 + 비밀번호 형식이 유효한 형식인 경우
+  else {
+    passwordInput.classList.remove(INVALID_CLASS);
   }
 
-  signUpButton.addEventListener("submit", handleSignUp);
+  // 비밀번호 확인 입력부분이 비어 있는 경우
+  if (!confirmPasswordValue.trim()) {
+    confirmPasswordError.textContent = "비밀번호를 입력해 주세요.";
+    confirmPasswordInput.classList.add(INVALID_CLASS);
+  }
+  // 비밀번호와 비밀번호 확인이 일치하지 않는 경우
+  else if (passwordValue !== confirmPasswordValue) {
+    confirmPasswordError.textContent = "비밀번호가 일치하지 않아요.";
+    confirmPasswordInput.classList.add(INVALID_CLASS);
+  }
+  // 비밀번호 확인 입력부분이 비어있지 않은 경우 + 비밀번호 형식이 유효한 형식인 경우
+  else {
+    confirmPasswordInput.classList.remove(INVALID_CLASS);
+  }
+}
+
+emailInput.addEventListener("blur", handleEmailInput);
+emailInput.addEventListener("input", handleEmailInput);
+passwordInput.addEventListener("blur", handlePasswordInput);
+passwordInput.addEventListener("input", handlePasswordInput);
+confirmPasswordInput.addEventListener("blur", handlePasswordInput);
+confirmPasswordInput.addEventListener("input", handlePasswordInput);
+
+function validateSignUp() {
+  return (
+    !emailError.textContent &&
+    !passwordError.textContent &&
+    !confirmPasswordError.textContent
+  );
+}
+
+async function handleSignUp(event) {
+  event.preventDefault();
+
+  if (!validateSignUp()) {
+    // 유효성 검사에 실패한 경우
+    return;
+  }
```

Comment:
주석이 조건문 위에 있어야 더 가독성이 좋을 것 같습니다. 조건에 대한 설명이라서요.

```suggestion
  // 유효성 검사에 실패한 경우
  if (!validateSignUp()) {
    return;
  }
```